altair/vegalite/v5/schema/_config.py

```diff
@@ -0,0 +1,1055 @@
+    strokeOpacity: float
+    strokeWidth: float
+    tension: float
+    text: str | Sequence[str]
+    theta: float
+    theta2: float
+    thickness: float
+    timeUnitBandPosition: float
+    timeUnitBandSize: float
+    tooltip: str | bool | None | float | TooltipContent
+    url: str
+    width: float
+    x: str | float
+    x2: str | float
+    y: str | float
+    y2: str | float
+
+
+class TitleConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    align: Align_T
+    anchor: TitleAnchor_T
+    angle: float
+    aria: bool
+    baseline: str | Baseline_T
+    color: str | None | ColorName_T
+    dx: float
+    dy: float
+    font: str
+    fontSize: float
+    fontStyle: str | FontStyle
+    fontWeight: FontWeight_T
+    frame: str | TitleFrame_T
+    limit: float
+    lineHeight: float
+    offset: float
+    orient: TitleOrient_T
+    subtitleColor: str | None | ColorName_T
+    subtitleFont: str
+    subtitleFontSize: float
+    subtitleFontStyle: str | FontStyle
+    subtitleFontWeight: FontWeight_T
+    subtitleLineHeight: float
+    subtitlePadding: float
+    zindex: float
+
+
+class FormatConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    normalizedNumberFormat: str
+    normalizedNumberFormatType: str
+    numberFormat: str
+    numberFormatType: str
+    timeFormat: str
+    timeFormatType: str
+
+
+class ViewConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    clip: bool
+    continuousHeight: float
+    continuousWidth: float
+    cornerRadius: float
+    cursor: Cursor_T
+    discreteHeight: float
+    discreteWidth: float
+    fill: str | None | ColorName_T
+    fillOpacity: float
+    opacity: float
+    step: float
+    stroke: str | None | ColorName_T
+    strokeCap: StrokeCap_T
+    strokeDash: Sequence[float]
+    strokeDashOffset: float
+    strokeJoin: StrokeJoin_T
+    strokeMiterLimit: float
+    strokeOpacity: float
+    strokeWidth: float
+
+
+# TODO: Non-`TypedDict` arg
+
+
+class ScaleInvalidDataConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    angle: str | ScaleInvalidDataShowAsValueangle
+    color: str | ScaleInvalidDataShowAsValuecolor
+    fill: str | ScaleInvalidDataShowAsValuefill
+    fillOpacity: str | ScaleInvalidDataShowAsValuefillOpacity
+    opacity: str | ScaleInvalidDataShowAsValueopacity
+    radius: str | ScaleInvalidDataShowAsValueradius
+    shape: str | ScaleInvalidDataShowAsValueshape
+    size: str | ScaleInvalidDataShowAsValuesize
+    stroke: str | ScaleInvalidDataShowAsValuestroke
+    strokeDash: str | ScaleInvalidDataShowAsValuestrokeDash
+    strokeOpacity: str | ScaleInvalidDataShowAsValuestrokeOpacity
+    strokeWidth: str | ScaleInvalidDataShowAsValuestrokeWidth
+    theta: str | ScaleInvalidDataShowAsValuetheta
+    x: str | ScaleInvalidDataShowAsValuex
+    xOffset: str | ScaleInvalidDataShowAsValuexOffset
+    y: str | ScaleInvalidDataShowAsValuey
+    yOffset: str | ScaleInvalidDataShowAsValueyOffset
```

Comment:
All of these will be using a `TypedDict`, related [`a8ac7ec` (#3536)](https://github.com/vega/altair/pull/3536/commits/a8ac7ec647f1c27ffbffe9c3d1d7f5f866659835)

Will be roughly:

```py
from typing import Generic, TypeVar
from typing_extensions import TypedDict

T = TypeVar("T")

class Value(TypedDict, Generic[T]):
    value: T
```

Then instead of creating a `TypedDict` per `ScaleInvalidDataShowAsValue...`, they become:

```py
class ScaleInvalidDataConfigKwds(TypedDict, total=False):
    """Placeholder doc."""

    angle: str | Value[float]
    color: str | Value[ColorName_T]
    fill: str | Value[ColorName_T]
    fillOpacity: str | Value[float]
    opacity: str | Value[float]
    radius: str | Value[float]
    shape: str | Value[str]
    size: str | Value[float]
    stroke: str | Value[ColorName_T]
    strokeDash: str | Value[Sequence[float]]
    strokeOpacity: str | Value[float]
    strokeWidth: str | Value[float]
    theta: str | Value[float]
    x: str | Value[str | float]
    xOffset: str | Value[float]
    y: str | Value[str | float]
    yOffset: str | Value[float]

```

Reply:
Most of this is finished now in [`a9c8b99` (#3536)](https://github.com/vega/altair/pull/3536/commits/a9c8b99cb09e64a81b18a833019d7b1fdde5465e)